cmd/kritis/signer/main.go

```diff
@@ -32,37 +32,43 @@ import (
 	yaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+const (
+	SignerMode_CheckAndSign  = "check-and-sign"
+	SignerMode_CheckOnly     = "check-only"
+	SignerMode_BypassAndSign = "bypass-and-sign"
+)
+
 func main() {
-	var image, pri_key_path, passphrase, pub_key_path, policy_path string
+	var image, pri_key_path, passphrase, pub_key_path, policy_path, mode string
 
+	flag.StringVar(&mode, "mode", "", "mode of operation, check-and-sign|check-only|bypass-and-sign")
 	flag.StringVar(&image, "image", "", "image url, e.g., gcr.io/foo/bar@sha256:abcd")
 	flag.StringVar(&pri_key_path, "private_key", "", "signer private key path, e.g., /dev/shm/key.pgp")
 	flag.StringVar(&passphrase, "passphrase", "", "passphrase for private key, if any")
 	flag.StringVar(&pub_key_path, "public_key", "", "public key path, e.g., /dev/shm/key.pub")
 	flag.StringVar(&policy_path, "policy", "", "vulnerability signing policy file path, e.g., /tmp/vulnz_signing_policy.yaml")
 	flag.Parse()
 
-	glog.Infof("image: %s, s_key: %s, policy: %s", image, pri_key_path, policy_path)
+	if mode != SignerMode_CheckAndSign && mode != SignerMode_BypassAndSign {
+		if mode == SignerMode_CheckOnly {
+			glog.Fatalf("Mode %s note supported yet.", mode)
+		} else {
+			glog.Fatalf("Unrecognized mode %s.", mode)
+		}
+	}
 
 	signerKey, err := ioutil.ReadFile(pri_key_path)
```

Comment:
This is only needed if signing is requested, so could you make sure we don't try to read the key in check-only?

Reply:
Will do this when implementing `check-only` mode, which is not being supported as part of this PR.